File: Python3/2302. Count Subarrays With Score Less Than K.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        ans = prefix = l = 0
        for r, n in enumerate(nums):
            prefix += n
            while l <= r and prefix * (r - l + 1) >= k:
                prefix -= nums[l]
                l += 1

            ans += r - l + 1

        return ans

File: Java/2302. Count Subarrays With Score Less Than K.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        int l = 0;
        long ans = 0, prefix = 0;
        for (int r = 0; r < nums.length; r++) {
            prefix += nums[r];
            while (l <= r && prefix * (r - l + 1) >= k) {
                prefix -= nums[l++];
            }
            ans += r - l + 1;
        }

        return ans;
    }
}